public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -493,6 +500,7 @@ public SurveyQuestionResult apply(AchillesResult o) {
             toMatchConceptIds.addAll(drugMatchedConcepts.stream().map(Concept::getConceptId).collect(Collectors.toList()));
         }
 
+        searchTrackEvent("Search", "DomainSearch", query);
```

Comment:
Does this really happen asynchronously? If so Spring is doing some insane black magic. From scanning through the docs I wasn't quite sure if this would work, or if it would need to be a call made to another `@Service` (or spring-instantiated object). Could you confirm from the logs?

(also can print `Thread.currentThread().getName()` here and in `searchTrackEvent` to confirm)

Reply:
It gives different thread names.

| qtp1262773598-18      
 | qtp1262773598-16      
| qtp1262773598-22

Reply:
Neat - thanks for checking.